File: CS50 2019/Python/pset6/vigenere.py
from cs50 import get_string
from sys import argv, exit

if len(argv) != 2:
    print("Usage: python vigenere.py k")
    exit(1)

for ch in argv[1]:
    if not ch.isalpha():
        print("Usage: python vigenere.py k")
        exit(1)
        
k = argv[1]

p = get_string("plaintext:  ")

print("ciphertext: ", end="")

j = 0

for ch in p:
    if not ch.isalpha():
        print(ch, end="")
        continue
    
    ascii_ = 65 if ch.isupper() else 97
    pi = ord(ch) - ascii_
    kj = ord(k[j % len(k)].upper()) - 65
    ci = (pi + kj) % 26
    j += 1
    
    print(chr(ci + ascii_), end="")

print()
exit(0)

File: CS50 2019/C/pset2/vigenere.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <ctype.h>

void shifr(string text, string key);
bool check(string text);

int main(int argc, string argv[])
{
    if (argc == 2 && check(argv[1]))
    {
        string key = argv[1];
        string text = get_string("plaintext:  ");
        shifr(text, key);
    }
    else
    {
        printf("Usage: ./caesar key\n");
        return 1;
    }
    return 0;
}

bool check(string text)
{
    for (int i = 0; i < strlen(text); i++)
    {
        if(!isalpha(text[i]))
        {
            return false;
        }
    }
    return true;
}

void shifr(string text, string key)
{
    printf("ciphertext: ");
    int kLen = strlen(key);
        for (int i = 0, j = 0, n = strlen(text); i < n; i++)
        {
            int ch = tolower(key[j % kLen]) - 'a'; 
            if (isupper(text[i]))
            {
                char a = 'A' + ((text[i] - 'A' + ch) % 26);
                printf("%c", a);
                j++;
            }
            else if (islower(text[i]))
            {
                char a = 'a' + ((text[i] - 'a' + ch) % 26);
                printf("%c", a);
                j++;
            }
            else
            {
                printf("%c", text[i]);
            }
        }
        printf("\n");
}
